lib/recommendation.rb

```diff
@@ -0,0 +1,79 @@
+# suggestion for optimising: only pass in property id rather than full object?
+module Recommendation
+	
+	# tenant_features: utilities and furniture are missing from tenants schema
+	TENANT_PROPERTY_FEATURES = ["mobility_aids",
+								"accessible_shower",
+								"car_parking", 
+								"lift_access"]	
+
+	def recommend_properties(properties)
+		
+		# filter for properties in same location of user interest
+		#properties = generate_candidates(properties)	
+	
+		recommended = Hash.new(0)
+
+		@user_features =  self.attributes.slice(*TENANT_PROPERTY_FEATURES)
```

Comment:
For non-global variables that you do not plan on using in another file, remove the "@".